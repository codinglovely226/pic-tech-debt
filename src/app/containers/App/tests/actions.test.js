/* eslint-disable */

import {
  LOAD_REPOS,
  LOAD_REPOS_SUCCESS,
  LOAD_REPOS_ERROR,
} from '../constants';

import {
  batchInsertDocument,
  changeForm,
  clearError,
  deleteDocument,
  displayRecord,
  displayTemplate,
  executeSearch,
  initSiteConfig,
  insertDocument,
  loadRepos,
  loginRequest,
  logout,
  lookupMacroTemplate,
  registerRequest,
  repoLoadingError,
  reposLoaded,
  requestError,
  sendingRequest,
  setAuthState,
  setTab,
  setUiModeDefault,
  setUiModeEditTemplate,
  storeKey,
  swUpdated,
  syncingPouch,
  updateActionBarSearchResults,
  updateDocument,
  updateCaseListSearchResults,
  updateMacro,
  updateNewRecordSearchResults,
  updateToxicologySource,
} from '../actions';

describe('App Actions', () => {
  const mockedDate = new Date(2018, 2, 4, 4, 59, 13, 805);
  global.Date = jest.fn(() => mockedDate);

  describe('loadRepos', () => {
    it('should return the correct type', () => {
      const expectedResult = {
        type: LOAD_REPOS,
      };

      expect(loadRepos()).toEqual(expectedResult);
    });
  });

  describe('reposLoaded', () => {
    it('should return the correct type and the passed repos', () => {
      const fixture = ['Test'];
      const username = 'test';
      const expectedResult = {
        type: LOAD_REPOS_SUCCESS,
        repos: fixture,
        username,
      };

      expect(reposLoaded(fixture, username)).toEqual(expectedResult);
    });
  });

  describe('repoLoadingError', () => {
    it('should return the correct type and the error', () => {
      const fixture = {
        msg: 'Something went wrong!',
      };
      const expectedResult = {
        type: LOAD_REPOS_ERROR,
        error: fixture,
      };

      expect(repoLoadingError(fixture)).toEqual(expectedResult);
    });
  });
  describe('App action creators', () => {
    it('batchInsertDocument', () => {
      expect(batchInsertDocument([{ name: 'Case 1' }, { name: 'Case 2' }])).toMatchSnapshot();
    });
    it('changeForm', () => {
      expect(changeForm({ username: 'New form user', password: 'New form password' })).toMatchSnapshot();
    });
    it('clearError', () => {
      expect(clearError()).toMatchSnapshot();
    });
    it('deleteDocument', () => {
      expect(deleteDocument(1)).toMatchSnapshot();
    });
    it('displayRecord', () => {
      expect(displayRecord(1)).toMatchSnapshot();
    });
    it('displayTemplate', () => {
      expect(displayTemplate(1)).toMatchSnapshot();
    });
    it('executeSearch', () => {
      expect(executeSearch()).toMatchSnapshot();
    });
    it('initSiteConfig', () => {
      expect(initSiteConfig()).toMatchSnapshot();
    });
    it('insertDocument', () => {
      expect(insertDocument({ name: 'Case 1' })).toMatchSnapshot();
    });
    it('loginRequest', () => {
      expect(loginRequest({ name: 'Test user' })).toMatchSnapshot();
    });
    it('logout', () => {
      expect(logout()).toMatchSnapshot();
    });
    it('lookupMacroTemplate', () => {
      expect(lookupMacroTemplate('macro')).toMatchSnapshot();
    });
    it('registerRequest', () => {
      expect(registerRequest({ name: 'Test request' })).toMatchSnapshot();
    });
    it('requestError', () => {
      expect(requestError({ name: 'Test error' })).toMatchSnapshot();
    });
    it('sendingRequest', () => {
      expect(sendingRequest(true)).toMatchSnapshot();
    });
    it('setAuthState', () => {
      expect(setAuthState(false)).toMatchSnapshot();
    });
    it('setTab', () => {
      expect(setTab(1)).toMatchSnapshot();
    });
    it('setUiModeDefault', () => {
      expect(setUiModeDefault()).toMatchSnapshot();
    });
    it('setUiModeEditTemplate', () => {
      expect(setUiModeEditTemplate()).toMatchSnapshot();
    });
    it('storeKey', () => {
      expect(storeKey('appKeyId', 'appKey')).toMatchSnapshot();
    });
    it('swUpdated', () => {
      expect(swUpdated()).toMatchSnapshot();
    });
    it('updateToxicologySource', () => {
      expect(updateToxicologySource()).toMatchSnapshot();
    });
    it('syncingPouch', () => {
      expect(syncingPouch(true)).toMatchSnapshot();
    });
    it('updateActionBarSearchResults', () => {
      expect(updateActionBarSearchResults(['Test Results'])).toMatchSnapshot();
    });
    it('updateCaseListSearchResults', () => {
      expect(updateCaseListSearchResults(['Test Results'])).toMatchSnapshot();
    });
    it('updateDocument', () => {
      expect(updateDocument({ name: 'Test Case' })).toMatchSnapshot();
    });
    it('updateMacro', () => {
      expect(updateMacro('macro')).toMatchSnapshot();
    });
    it('updateNewRecordSearchResults', () => {
      expect(updateNewRecordSearchResults(['Test Results'])).toMatchSnapshot();
    });
  });
});
