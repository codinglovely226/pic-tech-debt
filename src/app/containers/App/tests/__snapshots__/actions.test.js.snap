exports[`App Actions App action creators batchInsertDocument 1`] = `
Object {
  "documents": Array [
    Object {
      "name": "Case 1",
    },
    Object {
      "name": "Case 2",
    },
  ],
  "type": "pic/App/BATCH_INSERT_DOCUMENT",
}
`;

exports[`App Actions App action creators changeForm 1`] = `
Object {
  "newFormState": Object {
    "password": "New form password",
    "username": "New form user",
  },
  "type": "pic/App/CHANGE_FORM",
}
`;

exports[`App Actions App action creators clearError 1`] = `
Object {
  "type": "pic/App/CLEAR_ERROR",
}
`;

exports[`App Actions App action creators deleteDocument 1`] = `
Object {
  "id": 1,
  "type": "pic/App/DELETE_DOCUMENT",
}
`;

exports[`App Actions App action creators displayRecord 1`] = `
Object {
  "id": 1,
  "type": "pic/App/DISPLAY_RECORD",
}
`;

exports[`App Actions App action creators displayTemplate 1`] = `
Object {
  "templateId": 1,
  "type": "pic/App/DISPLAY_TEMPLATE",
}
`;

exports[`App Actions App action creators executeSearch 1`] = `
Object {
  "context": undefined,
  "criteria": undefined,
  "type": "pic/App/EXECUTE_SEARCH",
}
`;

exports[`App Actions App action creators initSiteConfig 1`] = `
Object {
  "configId": undefined,
  "siteId": undefined,
  "type": "pic/App/INIT_SITE_CONFIG",
}
`;

exports[`App Actions App action creators insertDocument 1`] = `
Object {
  "document": Object {
    "name": "Case 1",
  },
  "type": "pic/App/INSERT_DOCUMENT",
}
`;

exports[`App Actions App action creators loginRequest 1`] = `
Object {
  "data": Object {
    "name": "Test user",
  },
  "type": "pic/App/LOGIN_REQUEST",
}
`;

exports[`App Actions App action creators logout 1`] = `
Object {
  "type": "pic/App/LOGOUT",
}
`;

exports[`App Actions App action creators lookupMacroTemplate 1`] = `
Object {
  "macro": "macro",
  "type": "pic/App/LOOKUP_MACRO",
}
`;

exports[`App Actions App action creators registerRequest 1`] = `
Object {
  "data": Object {
    "name": "Test request",
  },
  "type": "pic/App/REGISTER_REQUEST",
}
`;

exports[`App Actions App action creators requestError 1`] = `
Object {
  "error": Object {
    "name": "Test error",
  },
  "type": "pic/App/REQUEST_ERROR",
}
`;

exports[`App Actions App action creators sendingRequest 1`] = `
Object {
  "sending": true,
  "type": "pic/App/SENDING_REQUEST",
}
`;

exports[`App Actions App action creators setAuthState 1`] = `
Object {
  "newAuthState": false,
  "token": undefined,
  "type": "pic/App/SET_AUTH",
}
`;

exports[`App Actions App action creators setTab 1`] = `
Object {
  "tab": 1,
  "type": "pic/App/SET_TAB",
}
`;

exports[`App Actions App action creators setUiModeDefault 1`] = `
Object {
  "type": "pic/App/SET_UI_MODE_DEFAULT",
}
`;

exports[`App Actions App action creators setUiModeEditTemplate 1`] = `
Object {
  "type": "pic/App/SET_UI_MODE_EDIT_TEMPLATES",
}
`;

exports[`App Actions App action creators storeKey 1`] = `
Object {
  "appKey": "appKey",
  "appKeyId": "appKeyId",
  "type": "pic/App/STORE_KEY",
}
`;

exports[`App Actions App action creators swUpdated 1`] = `
Object {
  "type": "pic/App/SW_UPDATED",
}
`;

exports[`App Actions App action creators syncingPouch 1`] = `
Object {
  "syncing": true,
  "type": "pic/App/SYNCING_POUCH",
}
`;

exports[`App Actions App action creators updateActionBarSearchResults 1`] = `
Object {
  "results": Array [
    "Test Results",
  ],
  "type": "pic/App/UPDATE_ACTION_BAR_SEARCH_RESULTS",
}
`;

exports[`App Actions App action creators updateCaseListSearchResults 1`] = `
Object {
  "results": Array [
    "Test Results",
  ],
  "type": "pic/App/UPDATE_CASE_LIST_SEARCH_RESULTS",
}
`;

exports[`App Actions App action creators updateDocument 1`] = `
Object {
  "document": Object {
    "name": "Test Case",
  },
  "type": "pic/App/UPDATE_DOCUMENT",
}
`;

exports[`App Actions App action creators updateMacro 1`] = `
Object {
  "macro": "macro",
  "type": "pic/App/UPDATE_MACRO",
}
`;

exports[`App Actions App action creators updateNewRecordSearchResults 1`] = `
Object {
  "results": Array [
    "Test Results",
  ],
  "type": "pic/App/UPDATE_NEW_RECORD_SEARCH_RESULTS",
}
`;

exports[`App Actions App action creators updateToxicologySource 1`] = `
Object {
  "toxicologySource": undefined,
  "type": "pic/App/TOXICOLOGY_SOURCE",
}
`;
